src/compas_assembly/datastructures/assembly.py

```diff
@@ -326,7 +326,9 @@ def interfaces(self):
 
         """
         for edge in self.graph.edges():
-            yield self.edge_interface(edge)
+            if (interfaces := self.edge_interfaces(edge)):
```

Comment:
this will not work in python 2.x

Reply:
I removed the walrus operator. now should be compatible